src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -0,0 +1,16 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+
+public interface VisionSubsystem {
+  public Pose3d getPose3d();
```

Comment:
These first 3 methods aren't used outside of their implementing classes, so they probably don't need to be part of the interface.

I also think that once you implement the refactor to remove the excess calls to `postEstimator.update()`, you will need to make these private for them to make any sense.